legal-api/src/legal_api/services/filings/validations/change_of_registration.py

```diff
@@ -39,6 +42,19 @@ def validate(business: Business, filing: Dict) -> Optional[Error]:
         return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid filing is required.')}])
 
     msg = []
+    for item in find_updated_keys_for_firms(business, filing, filing_type):
+        if flags.is_on('enabled-deeper-permission-action') and item.get('email_changed'):
+            required_permission = ListActionsPermissionsAllowed.FIRM_EDITABLE_EMAIL_ADDRESS.value
+            message = 'Permission Denied - You do not have permissions edit email in this filing.'
+            error = PermissionService.check_user_permission(required_permission, message=message)
+            if error:
+               return error
```

Comment:
I don't think this is correct. `FIRM_EDITABLE_DBA` should have precedence. If user has `FIRM_EDITABLE_DBA` permission then they should be able to modify email. This code check `FIRM_EDITABLE_EMAIL_ADDRESS` first and error out if user don't have this permission even though they have `FIRM_EDITABLE_DBA`